tests/testthat/test-archive.R

```diff
@@ -0,0 +1,24 @@
+library(dplyr)
+
+test_that("first input must be a data.frame",{
+  expect_error(as_epi_archive(c(1,2,3)))
+})
+
+dt <- archive_cases_dv_subset$DT
+
+test_that("data.frame must contain geo_value, time_value and version columns",{
+  expect_error(as_epi_archive(select(dt,-geo_value)))
+  expect_error(as_epi_archive(select(dt,-time_value)))
+  expect_error(as_epi_archive(select(dt,-version)))
+})
+
+test_that("other_keys can only contain names of the data.frame columns",{
+  expect_error(as_epi_archive(dt,other_keys = "xyz"))
+  expect_error(as_epi_archive(dt,other_keys = "percent_cli"),NA)
+})
+
+test_that("other_keys cannot contain names geo_value, time_value or version",{
+  expect_error(as_epi_archive(dt,other_keys = "geo_value"))
+  expect_error(as_epi_archive(dt,other_keys = "time_value"))
+  expect_error(as_epi_archive(dt,other_keys = "version"))
+})
```

Comment:
Please also add a test that the key of the archive's $DT is set properly, regardless of what kind of (valid) `x` arg it is given: a vanilla data frame, a tibble, a keyed data.table, an unkeyed data.table, and maybe even an epi_df.  See [here](https://github.com/dajmcdon/epiprocess/blob/74f989748a5f63d400d161608898a36f3eb6c7ce/R/archive.R#L168) how setting the key has some gotchas.  Probably good to test to make sure we don't break this again because I don't think we'd notice just from reading the code.

Reply:
"keyed data.table" isn't precise enough.  We need to test constructing an archive from an `x` arg that's a data.table with a key that's not the same as the one we want the $DT field of the archive to have.

Reply:
I did one where the DT's keys are different from what is passed on to the argument for `as_epi_archive`.

Reply:
Thanks for the extensive tests!  I think we might be missing one more case: an epi_df with nontrivial other_keys metadata.
- If we provide `other_keys` to `as_epi_archive`, it should override the `epi_df`'s `other_keys` metadata
- If we don't, we should use the `epi_df`'s `other_keys` metadata (so if the epi df had "code" as another key, we the archive by default should have key cols geo_value, time_value, code, version)

Reply:
I just found that `additional_metadata` isn't preserved.

Reply:
So it looks like we have a few unwanted behaviors:
- keyed data.table + no other_keys arg --> we ignore the current key & replace with default geo time version
- epi_df with other_keys + no other_keys arg --> same deal
- epi_df with additional metadata --> no $additional_metadata

Feel like tackling those here, or filing them as issues + commenting in the tests that this is /unwanted/ behavior that should change?

Reply:
As for the desired behavior regarding key-setting with no other_keys arg, I've described different things for the keyed data.table vs. the epi_df, but we might want to do roughly the same thing (this is different from the things I described before):
- if input's key includes geo_value and time_value, use `c(input_key[input_key != "version"], "version")` as the epi_archive key --- if they already had "version" as part of the key, this reorders it to be at the end of the key (some epi_archive operations rely on this ordering), and if "version" wasn't already part of the key, it adds it at the end.

Reply:
For the additional_metadata, I think could just
- use the `additional_metadata` arg if provided
- use the stuff from the epi_df's metadata that makes sense if they don't provide an `additional_metadata` arg

Reply:
Okay, I think we have this filed away in issues.  Not sure if the additional test here was added, but that could be part of those issues.